libtransmission/magnet.c

```diff
@@ -174,7 +174,7 @@ tr_magnet_info* tr_magnetParse(char const* uri)
                 }
             }
 
-            if (vallen > 0 && keylen == 2 && memcmp(key, "dn", 2) == 0)
+            if (vallen > 0 && keylen == 2 && memcmp(key, "dn", 2) == 0 && displayName == NULL)
```

Comment:
_(nit)_ a `NULL` ptr check is likely cheaper than a call to memcmp; maybe put it first in the conditional?

```suggestion
            if (displayName == NULL && vallen > 0 && keylen == 2 && memcmp(key, "dn", 2) == 0)
```